pkg/query/endpointset.go

```diff
@@ -656,17 +655,12 @@ type endpointRef struct {
 
 // newEndpointRef creates a new endpointRef with a gRPC channel to the given the IP address.
 // The call to newEndpointRef will return an error if establishing the channel fails.
-func (e *EndpointSet) newEndpointRef(ctx context.Context, spec *GRPCEndpointSpec) (*endpointRef, error) {
+func (e *EndpointSet) newEndpointRef(spec *GRPCEndpointSpec) (*endpointRef, error) {
 	var dialOpts []grpc.DialOption
 
 	dialOpts = append(dialOpts, e.dialOpts...)
 	dialOpts = append(dialOpts, spec.dialOpts...)
-	// By default DialContext is non-blocking which means that any connection
-	// failure won't be reported/logged. Instead block until the connection is
-	// successfully established and return the details of the connection error
-	// if any.
-	dialOpts = append(dialOpts, grpc.WithReturnConnectionError())
-	conn, err := grpc.DialContext(ctx, spec.Addr(), dialOpts...)
```

Comment:
I've only skimmed over the PR but does it mean that the process will still log useful information in case of TLS misconfiguration? If not, it would be a regression but not sure what can be done.

(I described a relatively easy reproducer in https://github.com/thanos-io/thanos/issues/4923#issuecomment-1582151553, not sure if it's still working)

Reply:
I think i could test and seems to be working just fine:

```
ts=2024-06-07T18:13:40.618655Z caller=endpointset.go:471 level=warn component=endpointset msg="update of endpoint failed" err="getting metadata: rpc error: code = DeadlineExceeded desc = latest balancer error: connection error: desc = \"error reading server preface: remote error: tls: expired certificate\"" address=0.0.0.0:10901
```

Reply:
Thanks a lot for verifying! All good then :)